Add analytics to page

impaktor opened the pull request:
It would be nice to see how much traffic we get to the site (and correlation with posts / announcements), and which countries (we used to be very "UK heavy", I suspect).

I already have an instance of [umami](https://umami.is/) running on the Pioneer box/VM, so I think embedding the tracking code is the last piece of the puzzle. 

I would have expected this PR to insert the tracking id in the header of each generated file, but grep tells me I'm not fully there (yet). Suggestions?

Code adapted from [chringel21](https://github.com/chringel21/chringel-hugo-theme/tree/main), and his [blog post](https://chringel.dev/2022/04/umami-a-google-analytics-alternative-for-hugo/)

impaktor commented on themes/pioneer/layouts/partials/analytics.html:
```diff
@@ -0,0 +1,11 @@
+{{ if and .Site.Config.Services.GoogleAnalytics.ID (not hugo.IsServer) }}
+{{ template "_internal/google_analytics.html" . }}
+{{/* Add support for umami website analytics */}}
+{{ else if and .Site.Params.umami.enabled (not hugo.IsServer) }}
```
OK, the error is in these conditional statements. It works if I remove them. 
Not that strange since those 4 lines are copy-pasted and I don't gork them.

impaktor commented:
I've updated the files, removed the conditional stuff in analytics.html, webpage seems to work, and looks like we get tracking on download, contribute, index pages:
```
 grep 795fff018383 -R *
config.toml:47:websiteId = "87e7c902-33ee-47d1-8ea1-795fff018383"
docs/page/download/index.html:29:  data-website-id="87e7c902-33ee-47d1-8ea1-795fff018383"
docs/page/contribute/index.html:29:  data-website-id="87e7c902-33ee-47d1-8ea1-795fff018383"
docs/page/donate/index.html:29:  data-website-id="87e7c902-33ee-47d1-8ea1-795fff018383"
docs/index.html:30:  data-website-id="87e7c902-33ee-47d1-8ea1-795fff018383"

```

sturnclaw commented on config.toml:
```diff
@@ -42,3 +42,7 @@ weight=15
+
+[Params.umami]
+websiteId = "87e7c902-33ee-47d1-8ea1-795fff018383"
+jsLocation = "http://191.101.80.52:3000/script.js"
```
This ideally shouldn't use the server IP directly but rather another CNAME e.g. `analytics.pioneerspacesim.net`. Additionally, it should probably be using SSL either through umami itself or more likely the nginx proxy already running for the forum etc.

sturnclaw commented:
Other than the above configuration tweak regarding the script source, looks good overall. Having analytics on site views (as well as possibly download button clicks from website) is probably a good idea in the long run.

impaktor commented:
- I've added the "CNAME" (wiki & forum used "A" instead of "CNAME", I'm hazy on what it should be) to analytics.pioneerspacesim.net
- I've added the following config (based on guesstimation and reading [random stuff](https://www.reddit.com/r/selfhosted/comments/12u4htl/umami_on_apache_reverse_proxy_ssl/)) and restarted the Apache2 server process (no errors):  
```html
<VirtualHost *:80>
        ServerName analytics.pioneerspacesim.net
        <Location />
                ProxyPreserveHost On
                ProxyPass http://localhost:3000
                ProxyPassReverse http://localhost:3000
        </Location>
        RemoteIPHeader X-Forwarded-For
        RequestHeader set $host X-Forwarded-Host
        RequestHeader set $proxy_add_x_forwarded_for X-Forwarded-For
        RequestHeader set $remote_addr X-Forwarded-For
</VirtualHost>

# <VirtualHost *:80>
#         ServerName analytics.pioneerspacesim.net
#         Redirect permanent / https://analytics.pioneerspacesim.net/
# </VirtualHost>

```
- I have not added the new sub-domain to the certbot. I assume this should also be done?
- (I have not pushed anything new to this PR yet, until I know it works, and I assume DNS records need time to update)

impaktor commented:
Okeli-dokeli, now everything works, I believe. Have set up https://analytics.pioneerspacesim.net, and created a pioneer team (I'll add users to it after merge), added certificate, and I've tested tracking visits to my [gitlab blog](https://impaktor.gitlab.io) and it works.

 (except I don't know why umami lost its memory from one restart to another. Seems like it didn't find the old data base and made a new one. I've tried several restarts now (as normal user), and it retains its memory, and uses the old data base).

impaktor commented:
If we want tracking of the API docs:
https://github.com/NaturalDocs/NaturalDocs/issues/96#issuecomment-2191756704

sturnclaw commented:
Not sure that we really need to add analytics to the code or developer docs in all honesty. The main website is fine, as is tracking downloads, but I think that's as far as we go for "useful" data.
